lib/context.ml

```diff
@@ -13,7 +13,12 @@ type t = Type.t MapOnKey.t
 
 let empty : t = MapOnKey.empty
 let find = Map.find
-let add key data = Map.add_exn ~key ~data
+let add key data t =
+  match Map.add ~key ~data t with
```

Comment:
Reading memo:
Using Map.add instead of Map.add_exn allows us to add an already inserted data to a map.